lib/artemis/client.rb

```diff
@@ -289,7 +333,8 @@ def respond_to_missing?(method_name, *_, &block) #:nodoc:
     # Internal collection object that holds references to the callback blocks.
     #
     # @api private
-    Callbacks = Struct.new(:before_callbacks, :after_callbacks) #:nodoc:
+    Callbacks = Struct.new(:before_execute_callbacks, :after_execute_callbacks,
+      :before_request_callbacks, :after_request_callbacks) #:nodoc:
```

Comment:
I think this `Callback` can stay as-is. Instead, the `Executor` should have `execute_callbacks` and `request_callbacks` and use them accordingly.

Reply:
I started out with that idea but changed it since it seemed a bit overkill:
- `Callbacks` -> `ExecuteCallbacks`
- `RequestCallbacks`

?